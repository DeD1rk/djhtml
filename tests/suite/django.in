<!-- Basic everyday Django template -->
{% load static %}
{% block content %}
{% if request.user.is_authenticated %}
<h1>Welcome, {{ request.user }}</h1>
{% else %}
<a href="{% url 'login' %}">Click here to login</a>
{% endif %}
{% endblock %}


<!-- Django tags within <style> -->
<style>
a {
{% block color %}
color: red;
{% endblock %}
}
</style>


<!-- Django tags within <script> -->
<script>
$(function() {
{% if verbose %}
console.log("Hi mom!");
{% endif %}
});
</script>


<!-- Django comment -->
{% comment %}
                   .------.
                   ||'||||'|
                  ,|| -.._ |
                  \|`.-. .-.
                   | `-'A'-'
                   | ( --.`|
                  _|`.`___,'__   ____
              _,-'_| _   _/   /';    `.
           ,-/ / '     /     ; ;       )
          / (__()     /     (__()\     |
         |  (__()    |   __(__() : `--  \
         \ _'__()`------'  (__() \       :
         ;  |\/|           |\/;\  \      |
         ;  '__'           |__| `. ;.__, (
         :   \ /           ;\;|   /  \    ,--.
         `._ \__\          ;__|  ;   |   /`.,-`.
         /   |\/|          |\;|  |   :   \`/`.__\
        ;    |__|___ __ ___|__;__|    \   `\`/`._`.
       ,     |\;|   _\/_   |\/|\/|     \  _|`,`.`.,--.
       ;    ;|__|___\/\/___|__|__|      `( ,/ /.`/`.,-`.
           ; |   ,   | |   `.    |       \`.;//;`\`/ ,. \
       jrei                  `-  |        `::_;   `\ `' /
                                                    `--'
{% endcomment %}


<!-- Django comment with optional comment -->
{% comment "Yo dawg" %}
            I heard you like comments
{% endcomment %}


<!-- Various spacings -->
{%comment    %}
             %}
{%    endcomment%}
{%if%}
{%block content    %}
Yuck!
{%    endblock%}
{%    endif    %}


<!-- Inconsistent nesting -->
{% block %}
1
{% with %}
2
{% endblock %}
1
{% endwith %}


<!-- Django tag {% plural %} should cause dedent-and-indent. Issue #24 -->
{% blocktrans count count=queryset.count %}
There is 1 item.
{% plural %}
There are {{ count }} items.
{% endblocktrans %}


<!-- Template tag directly following tag name. Issue #30 -->
<a{% if %} class=""{% endif %}>
Click here!
</a>


<!-- Multiline Jinja comment -->
{#
      ,-._|\
     /     .\
     \_,--._/
#}
