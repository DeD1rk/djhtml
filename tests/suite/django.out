<!-- Basic everyday Django template -->
{% load static %}
{% block content %}
    {% if request.user.is_authenticated %}
        <h1>Welcome, {{ request.user }}</h1>
    {% else %}
        <a href="{% url 'login' %}">Click here to login</a>
    {% endif %}
{% endblock %}


<!-- Django tags within <style> -->
<style>
    a {
        {% block color %}
            color: red;
        {% endblock %}
    }
</style>


<!-- Django tags within <script> -->
<script>
    $(function() {
        {% if verbose %}
            console.log("Hi mom!");
        {% endif %}
    });
</script>


<!-- Multiline comment -->
{% comment %}
      ,-._|\
     /     .\
     \_,--._/
{% endcomment %}


<!-- Multiline comment with optional comment -->
{% comment "Yo dawg" %}
            I heard you like comments
{% endcomment %}


<!-- Various spacings -->
{%comment    %}
             %}
{%    endcomment%}
{%if%}
    {%block content    %}
        Yuck!
    {%    endblock%}
{%    endif    %}


<!-- Inconsistent nesting -->
{% block %}
    1
    {% with %}
        2
    {% endblock %}
    1
{% endwith %}


<!-- Tags that both dedent and indent -->
{% blocktrans count count=queryset.count %}
    There is 1 item.
{% plural %}
    There are {{ count }} items.
{% endblocktrans %}


<!-- None of this should be indented -->
{% if %}
{% elif %}
{% else %}
{% endif %}
{% for %}
{% empty %}
{% endfor %}


<!-- Template tag directly following tag name -->
<a{% if %} class=""{% endif %}>
    Click here!
</a>


<!-- Ambiguous video tag from django-embed-video -->
{% video form.instance.video as my_video %}
    {% video my_video '360 x 200' %}
{% endvideo %}


<!-- Match the tag, the whole tag, and nothing but the tag -->
{% component_block "instruction" border_top=True header="Login" %}
    {% slot "body" %}
        {% with "login --token="|add:token as content %}
            {% component "copy_field" field_selector="token-field" content=content %}
        {% endwith %}
    {% endslot %}
{% endcomponent_block %}
